collect/collect.go

```diff
@@ -533,27 +566,47 @@ func (i *InMemCollector) sendExpiredTracesInCache(ctx context.Context, now time.
 
 	var totalSpansSent int64
 
+	toDelete := generics.NewSet[string]()
 	for _, t := range traces {
 		totalSpansSent += int64(t.DescendantCount())
 		_, span2 := otelutil.StartSpanWith(ctx, i.Tracer, "sendReadyTrace", "num_spans", int64(t.DescendantCount()))
-		var duration time.Duration
-		var reason string
 		if t.RootSpan != nil {
 			span2.SetAttributes(attribute.String("send_reason", TraceSendGotRoot))
-			duration, reason = i.send(ctx, t, TraceSendGotRoot)
+			td, err := i.makeDecision(t, TraceSendGotRoot)
+			if err != nil {
+				i.Logger.Error().WithFields(map[string]interface{}{
+					"trace_id": t.TraceID,
+				}).Logf("error making decision for trace: %s", err.Error())
+				continue
+			}
+			i.send(ctx, t, td)
+			toDelete.Add(t.TraceID)
```

Comment:
this might be updated if we decide to pass in a filter callback into the `TakeExpiredTraces`